private/bufpkg/bufimage/bufimagemodify/bufimagemodifyv2/bufimagemodifyv2.go

```diff
@@ -15,49 +15,165 @@
+}
+
+// PrefixSuffixOverride is an override that applies a suffix and a prefix.
+type PrefixSuffixOverride interface {
+	Override
+	GetPrefix() string
+	GetSuffix() string
+	prefixSuffixOverride()
+}
+
+// NewPrefxiSuffixOverride returns an override on both prefix and suffix.
+func NewPrefxiSuffixOverride(
+	prefix string,
+	suffix string,
+) PrefixSuffixOverride {
+	return newPrefixSuffixOverride(prefix, suffix)
 }
 
+// ValueOverride is an override that directly modifies a file option.
+type ValueOverride interface {
+	Override
+	valueOverride()
+}
+
+// NewValueOverride returns a new override on value.
+func NewValueOverride[T string | bool | descriptorpb.FileOptions_OptimizeMode](val T) ValueOverride {
+	return newValueOverride(val)
+}
+
+// ModifyOption is an option for ModifyXYZ.
+type ModifyOption func(*modifyOptions)
+
+// ModifyWithOverride modifies an option with override.
+func ModifyWithOverride(override Override) (ModifyOption, error) {
+	if override == nil {
+		return nil, errors.New("override must not be nil")
+	}
+	return func(options *modifyOptions) {
+		options.override = override
+	}, nil
+}
+
+type modifyOptions struct {
+	override Override
+}
+
+// ModifyJavaPackage modifies java_package. By default, it modifies java_package to
+// the file's proto package. Specify an override option to add prefix and/or suffix,
+// or set a value for this option.
 func ModifyJavaPackage(
 	marker Marker,
 	imageFile bufimage.ImageFile,
-	prefix string,
+	modifyJavaPackageOptions ...ModifyOption,
 ) error {
+	options := &modifyOptions{}
+	for _, option := range modifyJavaPackageOptions {
+		option(options)
+	}
+	var javaPackageValue string
+	switch t := options.override.(type) {
+	case prefixSuffixOverride:
+		javaPackageValue = getJavaPackageValue(imageFile, t.prefix, t.suffix)
+	case suffixOverride:
+		javaPackageValue = getJavaPackageValue(imageFile, "", t.Get())
+	case prefixOverride:
+		javaPackageValue = getJavaPackageValue(imageFile, t.Get(), "")
+	case valueOverride[string]:
+		javaPackageValue = t.get()
+	case nil:
+		javaPackageValue = getJavaPackageValue(imageFile, "", "")
+	default:
+		return fmt.Errorf("unknown Override type: %T", options.override)
+	}
+	if internal.IsWellKnownType(imageFile) {
+		return nil
+	}
 	descriptor := imageFile.Proto()
+	if descriptor.Options != nil && descriptor.Options.GetJavaPackage() == javaPackageValue {
+		// The option is already set to the same value, don't modify or mark it.
+		return nil
+	}
 	if descriptor.Options == nil {
 		descriptor.Options = &descriptorpb.FileOptions{}
 	}
-	descriptor.Options.JavaPackage = proto.String(getJavaPackageValue(imageFile, prefix))
+	descriptor.Options.JavaPackage = proto.String(javaPackageValue)
 	marker.Mark(imageFile, internal.JavaPackagePath)
 	return nil
 }
 
-func getJavaPackageValue(imageFile bufimage.ImageFile, prefix string) string {
+func getJavaPackageValue(imageFile bufimage.ImageFile, prefix string, suffix string) string {
```

Comment:
Maybe this can be moved to `bufgenv2` package? wdyt?
If this move happens, package `bufimagemodifyv2` won't need to know anything about prefix and suffix. `ModifyXYZ` will be take a value and modify the file option on descriptor set directly to this value. However, `bufgenv2` would need to do stuff like converting a image file's proto package to Pascal cas.